flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ArchivedExecutionGraph.java

```diff
@@ -351,4 +354,64 @@ public static ArchivedExecutionGraph createFrom(ExecutionGraph executionGraph) {
 			executionGraph.getCheckpointStatsSnapshot(),
 			executionGraph.getStateBackendName().orElse(null));
 	}
+
+	/**
+	 * Create ArchivedExecutionGraph from failed JobGraph.
+	 * Note: Some elements in the ArchivedExecutionGraph will be filled on a best-effort basis
+	 */
+	public static ArchivedExecutionGraph createFromFailedInit(
+		JobGraph jobGraph,
+		@Nullable Throwable throwable,
+		JobStatus finalJobStatus,
+		long initializationTimestamp) {
+		long failureTime = System.currentTimeMillis();
+		final int numberVertices = jobGraph.getNumberOfVertices();
+
+		Map<JobVertexID, ArchivedExecutionJobVertex> archivedTasks = new HashMap<>(numberVertices);
+		List<ArchivedExecutionJobVertex> archivedVerticesInCreationOrder = new ArrayList<>(numberVertices);
+		final Map<String, SerializedValue<OptionalFailure<Object>>> serializedUserAccumulators = Collections.emptyMap();
+		StringifiedAccumulatorResult[] archivedUserAccumulators = new StringifiedAccumulatorResult[]{};
+
+		final long[] timestamps = new long[JobStatus.values().length];
+		timestamps[JobStatus.FAILED.ordinal()] = failureTime;
+		timestamps[JobStatus.INITIALIZING.ordinal()] = initializationTimestamp;
+
+		String jsonPlan = "{}";
+		try {
+			jsonPlan = JsonPlanGenerator.generatePlan(jobGraph);
+		} catch (Throwable ignored) {}
+
+		ExecutionConfig ec;
+		try {
+			// note: this is using the system classloader, not the usercode classloader, thus it
+			// will fail in some cases. We can not access the usercode CL, when a JobManager initialization is detected.
+			ec = jobGraph.getSerializedExecutionConfig().deserializeValue(ArchivedExecutionGraph.class.getClassLoader());
```

Comment:
Maybe an empty `ExecutionConfig` is good enough.

Reply:
Or we have to pass in the user code class loader.

Reply:
Can we be certain that the user-code classloader could be created?

Reply:
For now, I'm creating an empty execution config.